proselint/tools.py

```diff
@@ -1,121 +1,185 @@
 """General-purpose tools shared across linting checks."""
 
 import json
-from io import FileIO
-from itertools import islice
+from collections.abc import Callable
+from itertools import accumulate, chain, islice
 from operator import itemgetter
-from typing import NamedTuple, Union
+from pathlib import Path
+from re import Pattern, finditer
+from re import compile as rcompile
+from sys import stdin
+from typing import Union, cast, overload
 
 from proselint.config import DEFAULT, Config
 from proselint.registry import CheckRegistry
 from proselint.registry.checks import LintResult
 
+ACCEPTED_EXTENSIONS = [".md", ".txt", ".rtf", ".html", ".tex", ".markdown"]
```

Comment:
```suggestion
ACCEPTED_EXTENSIONS = {".md", ".txt", ".rtf", ".html", ".tex", ".markdown"}
```
Any reason why it wasn't from the start?